src/docfx/build/page/BuildPage.cs

```diff
@@ -225,7 +225,7 @@ private static (object output, JObject extensionData) ApplyTemplate(Document fil
 
             if (file.Docset.Legacy)
             {
-                if (isPage)
+                if (isPage && file.Docset.Template != null)
```

Comment:
the condition is not really matched with below using item, not really straightforward.

Reply:
If put inside, the method signature does not match. When `Template == null`, it is currently best to return a `PageModel`, but `Template.Transform` currently returns a `TemplateModel`, so I'd rather keep the current change.

Reply:
`Docset.Template` is used inside of `TemplateTransform.Transform`, so it's better checked inside of `TemplateTransform.Transform`. For the signature, we can change the it right?

Reply:
just an example;
```C#
            if (file.Docset.Legacy)
            {
                if (isPage && TemplateTransform.TryTransform(model, file, out var templateModel, out var metadata))
                {
                    return (templateModel, metadata);
                }

                return (model, null);
            }
```
```C#
        public static bool TryTransform(PageModel pageModel, Document file, out TemplateModel model, out JObject metadata)
        {
            model = default;
            metadata = default;
            if (file.Docset.Template == null)
            {
                return false;
            }

            (model, metadata) = Transform(pageModel, file);
            return true;
        }
```

Reply:
offline discussed, add TODO here